service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/handlers/AdditionalApplicationsUploadedEventHandlerEmailTemplateTest.java

```diff
@@ -37,30 +42,59 @@
 import static uk.gov.hmcts.reform.fpl.enums.TabUrlAnchor.OTHER_APPLICATIONS;
 import static uk.gov.hmcts.reform.fpl.enums.UserRole.LOCAL_AUTHORITY;
 import static uk.gov.hmcts.reform.fpl.handlers.NotificationEventHandlerTestData.AUTH_TOKEN;
+import static uk.gov.hmcts.reform.fpl.handlers.NotificationEventHandlerTestData.LOCAL_AUTHORITY_NAME;
 import static uk.gov.hmcts.reform.fpl.testingsupport.email.EmailContent.emailContent;
 import static uk.gov.hmcts.reform.fpl.testingsupport.email.SendEmailResponseAssert.assertThat;
 import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.wrapElements;
 
 @ContextConfiguration(classes = {
     AdditionalApplicationsUploadedEventHandler.class, EmailNotificationHelper.class,
-    AdditionalApplicationsUploadedEmailContentProvider.class, CaseUrlService.class
+    AdditionalApplicationsUploadedEmailContentProvider.class, CaseUrlService.class,
+    RepresentativeNotificationService.class
+})
+@MockBeans({
+    @MockBean(FeatureToggleService.class), @MockBean(IdamClient.class), @MockBean(RequestData.class),
+    @MockBean(SendDocumentService.class), @MockBean(OthersService.class), @MockBean(OtherRecipientsInbox.class)
 })
 class AdditionalApplicationsUploadedEventHandlerEmailTemplateTest extends EmailTemplateTest {
     private static final long CASE_ID = 12345L;
     private static final String FAMILY_MAN_CASE_NUMBER = "FAM_NUM";
     private static final String CHILD_LAST_NAME = "Jones";
     private static final String RESPONDENT_LAST_NAME = "Smith";
 
+    public static final CaseData CASE_DATA = CaseData.builder()
+        .id(CASE_ID)
+        .caseLocalAuthority(LOCAL_AUTHORITY_NAME)
+        .familyManCaseNumber(FAMILY_MAN_CASE_NUMBER)
+        .hearingDetails(wrapElements(HearingBooking.builder()
+            .startDate(LocalDateTime.of(2020, 2, 20, 20, 20, 0))
+            .type(HearingType.CASE_MANAGEMENT)
+            .build()))
+        .respondents1(wrapElements(Respondent.builder()
+            .party(RespondentParty.builder().lastName(RESPONDENT_LAST_NAME).build())
+            .build()))
+        .children1(wrapElements(Child.builder()
+            .party(ChildParty.builder().dateOfBirth(LocalDate.now()).lastName(CHILD_LAST_NAME).build())
+            .build()))
+        .additionalApplicationsBundle(wrapElements(AdditionalApplicationsBundle.builder()
+            .c2DocumentBundle(C2DocumentBundle.builder().type(WITH_NOTICE).supplementsBundle(List.of()).build())
+            .otherApplicationsBundle(OtherApplicationsBundle.builder()
+                .applicationType(C1_WITH_SUPPLEMENT)
+                .supplementsBundle(List.of())
+                .build())
+            .build()))
+        .build();
+
     @Autowired
     private AdditionalApplicationsUploadedEventHandler underTest;
 
-    @MockBean
+    @Autowired
     private IdamClient idamClient;
```

Comment:
I don't think idam is required here, couldn't see a bean that used it. You could update the mock such that instead of doing `requestData.authorisation()`, you instead mock`requestData.userRoles()`. The reason it is passing right now is because in this code
```java 
       List<String> roles = new ArrayList<>(requestData.userRoles());
        if (!roles.containsAll(UserRole.HMCTS_ADMIN.getRoleNames())) {
```
roles is being set to an empty list so the condition is never false